src/Fixer/Strict/StrictTypesFixer.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Strict;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Jordi Boggiano <j.boggiano@seld.be>
+ * @author SpacePossum
+ */
+final class StrictTypesFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, Tokens $tokens)
+    {
+        // check if the declaration is already done
+        $searchIndex = $tokens->getNextMeaningfulToken(0);
+        if (null === $searchIndex) {
+            $this->insertSequence($tokens); // declaration not found, insert one
+
+            return;
+        }
+
+        $sequence = $this->getDeclareStrictTypeSequence();
+        $sequenceLocation = $tokens->findSequence($sequence, $searchIndex, null, false);
+        if (null === $sequenceLocation) {
+            $this->insertSequence($tokens); // declaration not found, insert one
+
+            return;
+        }
+
+        // check if the declaration is at the right location
+        reset($sequenceLocation);
+        $sequenceStartIndex = key($sequenceLocation);
+
+        end($sequenceLocation);
+        $sequenceEndIndex = $tokens->getNextMeaningfulToken(key($sequenceLocation));
+
+        if (1 === $sequenceStartIndex) {
+            // declaration already at the correct location, fix CS only
+            $this->fixCodeStyleOfSequence($tokens, $sequenceEndIndex);
+            $this->fixWhiteSpaceAroundSequence($tokens, $sequenceEndIndex);
+
+            return;
+        }
+
+        // Handle end of statement of the sequence; i.e. semicolon vs. close tag. Remove a semicolon as well,
+        // but don't remove a close tag since comment placement might cause invalid code to be created.
+        if (!$tokens[$sequenceEndIndex]->isGivenKind(T_CLOSE_TAG)) {
+            $sequenceLocation[$sequenceEndIndex] = $tokens[$sequenceEndIndex];
+        }
+
+        foreach ($sequenceLocation as $index => $token) {
+            $tokens->clearTokenAndMergeSurroundingWhitespace($index);
+        }
+
+        $this->insertSequence($tokens);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription()
+    {
+        return 'Force strict types declaration in all files.';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // must ran before SingleBlankLineBeforeNamespaceFixer, NoBlankLinesBeforeNamespaceFixer, NoExtraConsecutiveBlankLinesFixer, NoWhitespaceInBlankLinesFixer
+        return 1;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return PHP_VERSION_ID >= 70000 && $tokens[0]->isGivenKind(T_OPEN_TAG);
```

Comment:
So, if you want to fix non-monolithic files, why do you require them to start with open tag ?

Currently, this won't be fixed:
```php
HTML
<?php
function foo () {}
?>
HTML
```

But this will:
```php
<?php
function foo () {}
?>
HTML
<?php
function foo2 () {}
?>
HTML
```